src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/AuthorizationApiServiceImpl.scala

```diff
@@ -86,9 +88,12 @@ final case class AuthorizationApiServiceImpl(
   private def upsertTenantBySelfcareId(selfcareId: String)(implicit contexts: Seq[(String, String)]): Future[UUID] =
     for {
       partyInstitution <- partyProcess.getInstitution(selfcareId)
+      _                <- Future
```

Comment:
⚠️ this check should not be performed here.

Use case:
1. Organization `P1` is in the allow list
2. Get from tenant management fails (not there yet)
3. `upsertTenantBySelfcareId` is invoked, `P1` is correctly verified to be allowed, the tenant is created and token is exchanged -> `P1` is allowed ✅ 
4. `P1` is now removed from the allow list
5. Get from tenant management succeed (previously created) -> `P1` is allowed ⚠️

Reply:
Hmm, I haven't thought about it indeed, probably we have to check at EVERY token exchange sadly.

Reply:
This means that we need to execute a party request for each session token.

Reply:
we could:
- retrieve from the tenant management using the selfcare id
- if found
  - if origin == IPA -> 🟢 
  - if origin != IPA, check if it is in the allow list (🟢 / 🔴 )
- if not found
  - retrieve from selfcare 
  - if origin == IPA -> 🟢 and upsert
  - if origin != IPA, check if it is in the allow list (🟢 and upsert / 🔴 )

this way selfcare requests should remain the same.

Does it make sense?